### ISSUE TITLE
Enums and interfaces do not have definitions in pop-ups

### ISSUE THREAD

**Issue by LAITONEN** [Microsoft/TypeScript#25894]
The issue might be related to [this one](https://github.com/Microsoft/TypeScript/issues/24409), but mine is not about version 3.0, and is not solely about imports.

## Search Terms

enum, description, definition, pop-up, tooltip, vs code

## Suggestion

Hovering over enums and interfaces, should show their definition, i.e. I have this enum (or an interface):
```
enum enumName {
    ONE = 1;
    TWO = 2;
    THREE = 3;
}
```

If I hover over this enum, wherever I use it (whether it is imported to another file or used locally where it is declared), I would like to see what this enum is all about (essentially what are the properties).

This is what I see right now when I hover over enums or interfaces:
![image](https://user-images.githubusercontent.com/23742868/43146584-303df7c0-8f6a-11e8-8e23-6ba8644428b4.png)
![image](https://user-images.githubusercontent.com/23742868/43146615-46645468-8f6a-11e8-8c05-5a5265b43162.png)

As a workaround, I currently need to add this type of JSDoc comments right above the enum declaration.

```
/** Props:
*    @param ONE 1;
*    @param TWO 2;
*    @param THREE 3;
*/
```

Which yields to:
![image](https://user-images.githubusercontent.com/23742868/43146843-883b7d08-8f6a-11e8-81a9-eb9b4bcfc1df.png)

## Use Cases

This feature would save time & effort.
The shortcoming of the current approach is that either need to add `${(2+numberOfEnumProperties)} lines` of comments or go to the file, where enum is declared, locate the enum there and then familiarize myself with its properties.

## Examples

I would like to see the declaration of an enum onHover, pretty much the same way I can see the declaration of a type or interface when I perform identical action.

## Checklist
I am not that experienced to be able to answer the questions below.
My suggestion meets these guidelines:
* [ ] This wouldn't be a breaking change in existing TypeScript / JavaScript code
* [ ] This wouldn't change the runtime behavior of existing JavaScript code
* [ ] This could be implemented without emitting different JS based on the types of the expressions
* [ ] This isn't a runtime feature (e.g. new expression-level syntax)

**Comment by LAITONEN** [comment]
Also, if I have interface, which extends another interface:
```
interface One {
    one: number;
}

interface Two extends One {
    two: string;
}
```

And I use `interface Two` somewhere and hover over it, I will get the same dull description as I got on the second screenshot in the initial post with interfaceName.

**Comment by mhegazy** [comment]
I agree. hovering over a type name to get a pop-up with the same text is rather useless. we can do better.

**Comment by ro-savage** [comment]
Just adding on to this. I find it really annoying that the type information for interfaces and enums are basically useless.

It also affects all the IDEs when they display the definitions. See this one for JetBrains IDE's: https://youtrack.jetbrains.com/oauth?state=%2Fissue%2FWEB-44531

**Comment by ro-savage** [comment]
Could someone point out where in the code base these definitions are created, and especially where the Types one are vs the Interface.

Then maybe someone from the community can take a look at actioning this.

**Comment by orta** [comment]
My tactic for figuring out stuff like this is to enable TSServer logging, perform the action you're interested in, then see what requests were made:

```
    {"seq":270,"type":"request","command":"quickinfo","arguments":{"file":"/Users/ortatherox/dev/typescript/new-website/packages/sandbox/src/typeAcquisition.ts","line":15,"offset":15}}
```

So this request is a `quickinfo` - a search I the codebase finds `getQuickInfoAtPosition - which is where the code starts - my bet, the code you're looking for is in `SymbolDisplay.getSymbolDisplayPartsDocumentationAndSymbolKind`.

**Comment by ugoi** [comment]
Any updates on this?
